platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/topology/StaticTopology.java

```diff
@@ -32,35 +31,33 @@
 public class StaticTopology implements NetworkTopology {
     private static final long SEED = 0;
 
-    private final NodeId selfId;
+    private final List<NodeId> peerNodes;
+
     /**
-     * Two nodes are neighbors if their indexes in the address book are neighbors in the connection graph.
+     * Two nodes are neighbors if their node indexes are neighbors in the connection graph.
      */
-    private final AddressBook addressBook;
-
     private final RandomGraph connectionGraph;
 
+    private final int selfIndex;
+
     /**
      * Constructor.
      *
      * @param random            a source of randomness, used to chose random neighbors, does not need to be
      *                          cryptographically secure
-     * @param addressBook       the current address book
-     * @param selfId            the ID of this node
+     * @param peers             the set of peers in the network
+     * @param selfIndex         the index of this node in the address book
      * @param numberOfNeighbors the number of neighbors each node should have
      */
     public StaticTopology(
             @NonNull final Random random,
-            @NonNull final AddressBook addressBook,
-            @NonNull final NodeId selfId,
+            @NonNull final List<PeerInfo> peers,
+            final int selfIndex,
             final int numberOfNeighbors) {
-        this.addressBook = Objects.requireNonNull(addressBook, "addressBook must not be null");
-        this.selfId = Objects.requireNonNull(selfId, "selfId must not be null");
-        this.connectionGraph = new RandomGraph(random, addressBook.getSize(), numberOfNeighbors, SEED);
-
-        if (!addressBook.contains(selfId)) {
-            throw new IllegalArgumentException("Address book does not contain selfId");
-        }
+        this.peerNodes =
+                Objects.requireNonNull(peers.stream().map(PeerInfo::nodeId).toList(), "peers must not be null");
```

Comment:
The `Objects.requireNonNull()` is not necessary. If `peers` is null then `peers.stream()` will throw an exception first.

Reply:
good point.

Reply:
change not made in latest commit

Reply:
resolving, as this is now outdated.